--- trunk/.config.init
+++ trunk/.config.init
@@ -1,2 +1,4 @@
 # Use "make defconfig" to expand this to a full .config
+CONFIG_TARGET_brcm63xx=y
+CONFIG_TARGET_brcm63xx_generic_AGPF_S0=y
 
@@ -84,3 +86,3 @@ CONFIG_PACKAGE_kmod-ath9k-common=m
 CONFIG_PACKAGE_kmod-ath9k-htc=m
-CONFIG_PACKAGE_kmod-b43=m
+CONFIG_PACKAGE_kmod-b43=y
 CONFIG_PACKAGE_kmod-brcmfmac=m
--- trunk/target/linux/brcm63xx/base-files/lib/brcm63xx.sh
+++ trunk/target/linux/brcm63xx/base-files/lib/brcm63xx.sh
@@ -81,6 +81,11 @@
 		status_led="AR1004G:green:power"
 		brcm63xx_has_reset_button="true"
 		;;
+	AGPF)
+		status_led="AGPF-S0:green:power"
+		brcm63xx_has_reset_button="true"
+		ifname=eth1
+		;;
 	AW4139 |\
 	AW4339U)
 		status_led="dsl-274xb:green:power"
--- trunk/target/linux/brcm63xx/base-files/etc/uci-defaults/02_network
+++ trunk/target/linux/brcm63xx/base-files/etc/uci-defaults/02_network
@@ -47,7 +47,6 @@
 96348GW-A |\
 96358VW |\
 96358VW2 |\
-AGPF-S0 |\
 "CPVA502+" |\
 DV201AMR |\
 DWV-S0 |\
@@ -59,9 +58,24 @@
 	ucidef_add_switch "eth1" "1" "1"
 	ucidef_add_switch_vlan "eth1" "1" "0 1 2 3 4 5"
 	;;
+	
+AGPF-S0)
+	ucidef_set_interfaces_lan_wan "eth1.1" "eth1.2"
+	ucidef_add_switch "eth1" "1" "1"
+	ucidef_add_switch_vlan "eth1" "1" "0 1 2 5t"
+	ucidef_add_switch "eth1" "1" "2"
+	ucidef_add_switch_vlan "eth1" "2" "3 5t"
+	;;
+
+HW553)
+	ucidef_set_interfaces_lan_wan "eth0.1" "eth0.2"
+	ucidef_add_switch "eth0" "1" "1"
+	ucidef_add_switch_vlan "eth1" "1" "0 1 2 5t"
+	ucidef_add_switch "eth0" "1" "2"
+	ucidef_add_switch_vlan "eth1" "2" "3 5t"
+	;;
 
 AW4139 |\
-HW553 |\
 HW556*)
 	ucidef_set_interface_lan "eth0.1"
 	ucidef_add_switch "eth0" "1" "1"
--- trunk/target/linux/brcm63xx/base-files/etc/uci-defaults/01_leds
+++ trunk/target/linux/brcm63xx/base-files/etc/uci-defaults/01_leds
@@ -11,17 +11,30 @@
 	ucidef_set_led_netdev "lan" "LAN" "A4001N1:green:eth" "eth0"
 	ucidef_set_led_usbdev "usb" "USB" "A4001N1:green:3g" "1-1"
 	;;
+	
 96328dg2x2)
 	ucidef_set_led_usbdev "usb" "USB" "A4001N:green:usb" "1-1"
 	;;
+	
 'F@ST2704V2')
 	ucidef_set_led_usbdev "usb" "USB" "F@ST2704V2:green:usb" "1-1"
 	;;
+	
+AGPF-S0)
+	ucidef_set_led_default "power_blue" "Power Status" "AGPF-S0:green:power" "1"
+	ucidef_set_led_netdev "wifi_green" "WIFI - RXTX" "AGPF-S0:green:wifi" "wlan0"
+	ucidef_set_led_netdev "wan_green" "WAN - RXTX" "AGPF-S0:green:internet" "eth1.1"
+	ucidef_set_led_usbdev "usb" "USB - Status" "AGPF-S0:green:service" "1-1"
+	;;
+	
 HW553)
-	ucidef_set_led_netdev "lan" "LAN" "HW553:blue:lan" "eth0"
-	ucidef_set_led_usbdev "usb1" "USB1" "HW553:red:hspa" "1-1"
-	ucidef_set_led_usbdev "usb2" "USB2" "HW553:blue:hspa" "1-2"
+	ucidef_set_led_default "power_blue" "Power Status" "HW553:blue:power" "1"
+	ucidef_set_led_netdev "wifi_red" "WIFI - RXTX" "HW553:red:wifi" "wlan0"
+	ucidef_set_led_netdev "wan_blue" "WAN - RXTX" "HW553:blue:lan" "eth0.2"
+	ucidef_set_led_usbdev "usb1" "USB1 - Status" "HW553:red:hspa" "1-1"
+	ucidef_set_led_usbdev "usb2" "USB2 - Status" "HW553:blue:hspa" "1-2"
 	;;
+	
 HW556*)
 	ucidef_set_led_netdev "lan" "LAN" "HW556:red:dsl" "eth0"
 	ucidef_set_led_usbdev "usb" "USB" "HW556:red:hspa" "1-2"

--- trunk/target/linux/brcm63xx/image/Makefile
+++ trunk/target/linux/brcm63xx/image/Makefile
@@ -191,20 +191,7 @@
 endef
 
 define Image/Build
-	dd if=$(KDIR)/root.$(1) of=$(BIN_DIR)/$(IMG_PREFIX)-root.$(1) bs=128k conv=sync
-
-	# BT Voyager V2500V, V2500V_SIP_CLUB, V2500V_AA
-	$(call Image/Build/CFE,$(1),V2500V_BB,6348,BTV2500V,,--layoutver 5)
-	# RTA1025W_16 (numerous routers)
-	$(call Image/Build/CFE,$(1),RTA1025W_16,6348,RTA1025W_16,,--layoutver 5)
-	# RTA1320_16M (numerous routers)
-	$(call Image/Build/CFE,$(1),RTA1320_16M,6338,RTA1320_16M,,--layoutver 5)
-
-	# Inventel Livebox
-	$(call Image/Build/RedBoot,livebox)
-
-	# Netgear CVG834G
-	$(call Image/Build/HCS,$(1),cvg834g,a020,0001,0022,$(KDIR)/vmlinux)
+	dd if=$(KDIR)/root.$(1) of=$(BIN_DIR)/$(IMG_PREFIX)-root.$(1) bs=128k conv=sync	
 
 	# build dtb image list
 	$(foreach image,$(TARGET_$(PROFILE)_IMAGES),$(call Image/Build/$(image),$(1)))	
--- trunk/files/etc/build
+++ trunk/files/etc/build
@@ -0,0 +1,1 @@
+Compiled by hitech95
--- trunk/package/network/services/hostapd/files/wps-hotplug.sh
--- trunk/package/network/services/hostapd/files/wps-hotplug.sh
@@ -1,8 +1,11 @@
 #!/bin/sh
 
 if [ "$ACTION" = "pressed" -a "$BUTTON" = "wps" ]; then
+	echo "255" > /sys/class/leds/AGPF-S0:red:service/brightness
 	for dir in /var/run/hostapd*; do
 		[ -d "$dir" ] || continue
 		hostapd_cli -p "$dir" wps_pbc
 	done
+	sleep 10
+	echo "0" > /sys/class/leds/AGPF-S0:red:service/brightness
 fi
