--- a/modules/luci-mod-admin-full/luasrc/model/cbi/admin_network/dhcp.lua
+++ b/modules/luci-mod-admin-full/luasrc/model/cbi/admin_network/dhcp.lua
@@ -224,7 +224,9 @@ s = m:section(TypedSection, "host", translate("Static Leases"),
 		"only hosts with a corresponding lease are served.") .. "<br />" ..
 	translate("Use the <em>Add</em> Button to add a new lease entry. The <em>MAC-Address</em> " ..
 		"indentifies the host, the <em>IPv4-Address</em> specifies to the fixed address to " ..
-		"use and the <em>Hostname</em> is assigned as symbolic name to the requesting host."))
+		"use and the <em>Hostname</em> is assigned as symbolic name to the requesting host. " ..
+		"The optional <em>Lease time</em> can be used to set non-standard host-specific " ..
+		"lease time, e.g. 12h, 3d or infinite."))
 
 s.addremove = true
 s.anonymous = true
@@ -241,6 +243,9 @@ mac.rmempty  = true
 ip = s:option(Value, "ip", translate("<abbr title=\"Internet Protocol Version 4\">IPv4</abbr>-Address"))
 ip.datatype = "or(ip4addr,'ignore')"
 
+time = s:option(Value, "leasetime", translate("Lease time"))
+time.rmempty  = true
+
 hostid = s:option(Value, "hostid", translate("<abbr title=\"Internet Protocol Version 6\">IPv6</abbr>-Suffix (hex)"))
 
 sys.net.arptable(function(entry)
--- a/modules/luci-mod-admin-full/luasrc/view/admin_system/flashops.htm
+++ b/modules/luci-mod-admin-full/luasrc/view/admin_system/flashops.htm
@@ -44,6 +44,7 @@ $Id$
 				</div>
 				<% end %>
 			</div>
+			<div class="cbi-section-descr"><%:Note: This router's firmware has been modified to support the 'reset' button located in the bottom of the Router. If you press the 'reset' button, the router resets itself to the initial settings (by removing the jffs partition completely and thus reverting to the original configuration files included in the firmware). If you want to disable this reset feature, delete the file '/etc/hotplug.d/button/05-reset'.%></div>
 			<br />
 			<div class="cbi-section-descr"><%:To restore configuration files, you can upload a previously generated backup archive here.%></div>
 			<div class="cbi-section-node">