--- trunk/.config.init
+++ trunk/.config.init
@@ -1,4 +1,6 @@
 # Use "make defconfig" to expand this to a full .config
+CONFIG_TARGET_brcm63xx=y
+CONFIG_TARGET_brcm63xx_generic_HG553=y
 
 # Per-package build logs in <buildroot>/logs
 CONFIG_DEVEL=y
--- trunk/target/linux/brcm63xx/image/Makefile
+++ trunk/target/linux/brcm63xx/image/Makefile
@@ -191,20 +191,7 @@
 endef
 
 define Image/Build
-	dd if=$(KDIR)/root.$(1) of=$(BIN_DIR)/$(IMG_PREFIX)-root.$(1) bs=128k conv=sync
-
-	# BT Voyager V2500V, V2500V_SIP_CLUB, V2500V_AA
-	$(call Image/Build/CFE,$(1),V2500V_BB,6348,BTV2500V,,--layoutver 5)
-	# RTA1025W_16 (numerous routers)
-	$(call Image/Build/CFE,$(1),RTA1025W_16,6348,RTA1025W_16,,--layoutver 5)
-	# RTA1320_16M (numerous routers)
-	$(call Image/Build/CFE,$(1),RTA1320_16M,6338,RTA1320_16M,,--layoutver 5)
-
-	# Inventel Livebox
-	$(call Image/Build/RedBoot,livebox)
-
-	# Netgear CVG834G
-	$(call Image/Build/HCS,$(1),cvg834g,a020,0001,0022,$(KDIR)/vmlinux)
+	dd if=$(KDIR)/root.$(1) of=$(BIN_DIR)/$(IMG_PREFIX)-root.$(1) bs=128k conv=sync	
 
 	# build dtb image list
 	$(foreach image,$(TARGET_$(PROFILE)_IMAGES),$(call Image/Build/$(image),$(1)))	
--- trunk/files/etc/build
+++ trunk/files/etc/build
@@ -0,0 +1,1 @@
+Compiled by hitech95
--- trunk/package/network/services/hostapd/files/wps-hotplug.sh
--- trunk/package/network/services/hostapd/files/wps-hotplug.sh
@@ -1,8 +1,11 @@
 #!/bin/sh
 
 if [ "$ACTION" = "pressed" -a "$BUTTON" = "wps" ]; then
+	echo "255" > /sys/class/leds/AGPF-S0:red:service/brightness
 	for dir in /var/run/hostapd*; do
 		[ -d "$dir" ] || continue
 		hostapd_cli -p "$dir" wps_pbc
 	done
+	sleep 10
+	echo "0" > /sys/class/leds/AGPF-S0:red:service/brightness
 fi
