--- trunk/.config.init
+++ trunk/.config.init
@@ -1,4 +1,6 @@
 # Use "make defconfig" to expand this to a full .config
+CONFIG_TARGET_lantiq=y
+CONFIG_TARGET_lantiq_xway_DGN3500=y
 
 # Per-package build logs in <buildroot>/logs
 CONFIG_DEVEL=y
@@ -162,7 +164,7 @@
 CONFIG_PACKAGE_ATH_DFS=y
 CONFIG_PACKAGE_kmod-ath10k=m
 CONFIG_PACKAGE_kmod-ath5k=m
-CONFIG_PACKAGE_kmod-ath9k=m
+CONFIG_PACKAGE_kmod-ath9k=y
 CONFIG_PACKAGE_kmod-ath9k-common=m
 CONFIG_PACKAGE_kmod-ath9k-htc=m
 CONFIG_PACKAGE_kmod-b43=m
--- trunk/target/linux/lantiq/dts/DGN3500.dtsi
+++ trunk/target/linux/lantiq/dts/DGN3500.dtsi
@@ -150,34 +150,42 @@
 	};
 
 	gpio-leds {
-		compatible = "gpio-leds";
+ 		compatible = "gpio-leds";
+ 		dsl {
+			label = "DGN3500:dsl";
+ 			gpios = <&gpio 4 1>;
+ 		};
 		internet {
-			label = "internet";
-			gpios = <&gpio 2 1>;
-		};
-		dsl {
-			label = "dsl";
-			gpios = <&gpio 4 1>;
-		};
+			label = "DGN3500:green:internet";
+ 			gpios = <&gpio 2 1>;
+ 		};
+		internet2 {
+			label = "DGN3500:red:internet";
+ 			gpios = <&gpio 30 1>;
+ 		}; 		
+ 		power {
+			label = "DGN3500:green:power";
+ 			gpios = <&gpio 34 1>;
+ 		};
+ 		power2 {
+			label = "DGN3500:red:power";
+ 			gpios = <&gpio 39 1>;
+ 		};
 		usb {
-			label = "usb";
-			gpios = <&gpio 22 1>;
-		};
-		power {
-			label = "power";
-			gpios = <&gpio 34 1>;
-		};
-		power2 {
-			label = "power2";
-			gpios = <&gpio 39 1>;
-		};
-		wifi {
-			label = "wifi";
-			gpios = <&gpio 51 1>;
-		};
-		wps {
-			label = "wps";
-			gpios = <&gpio 52 1>;
-		};
-	};
+			label = "DGN3500:usb";
+ 			gpios = <&gpio 22 1>;
+ 		};
+ 		wifi {
+			label = "DGN3500:green:wifi";
+ 			gpios = <&gpio 14 1>;
+ 		};
+		wifi2 {
+			label = "DGN3500:red:wifi";
+ 			gpios = <&gpio 51 1>;
+ 		};
+ 		wps {
+			label = "DGN3500:wps";
+ 			gpios = <&gpio 52 1>;
+ 		};
+ 	};
 };
--- trunk/target/linux/lantiq/base-files/sbin/dsl_notify.sh
+++ trunk/target/linux/lantiq/base-files/sbin/dsl_notify.sh
@@ -18,13 +18,13 @@
 
 local default
 config_load system
-config_get default led_adsl default
+config_get default led_dsl default
 if [ "$default" != 1 ]; then
 	case "$DSL_INTERFACE_STATUS" in
-	  "HANDSHAKE")  led_timer dsl 500 500;;
-	  "TRAINING")   led_timer dsl 200 200;;
-	  "UP")		led_on dsl;;
-	  *)		led_off dsl
+	  "HANDSHAKE")  led_timer led_dsl 500 500;;
+	  "TRAINING")   led_timer led_dsl 200 200;;
+	  "UP")		led_on led_dsl;;
+	  *)		led_off led_dsl
 	esac
 fi
--- trunk/target/linux/lantiq/base-files/etc/uci-defaults/02_network
+++ trunk/target/linux/lantiq/base-files/etc/uci-defaults/02_network
@@ -119,12 +119,67 @@
 	uci set network.lan.type='none'
 	;;
 
-DGN3500*)
+DGN3500)
 	lan_mac=$(mtd_get_mac_ascii uboot-env ethaddr)
 	wan_mac=$(macaddr_add "$lan_mac" 1)
 	ucidef_set_interface_lan "eth0.1"
 	ucidef_add_switch "switch0" "1" "1"
-	ucidef_add_switch_vlan "switch0" "1" "3 2 1 0 5t"
+	ucidef_add_switch_vlan "switch0" "1" "0 1 2 3 5t"
+	# Blinkrate: 0=43ms; 1=84ms; 2=120ms; 3=170ms; 4=340ms; 5=670ms
+	uci set network.@switch[-1].blinkrate='1'
+
+	ucidef_add_switch_port "switch0" "1"
+	# Port 1 controls the GREEN configuration of LEDs for
+	# the switch and the section does not correspond to a real
+	# switch port.
+	#
+	# 0=LED off; 1=Collision/FDX; 2=Link/activity; 3=1000 Mb/s;
+	# 4=100 Mb/s; 5=10 Mb/s; 6=1000 Mb/s+activity; 7=100 Mb/s+activity;
+	# 8=10 Mb/s+activity; 9=10/100 Mb/s+activity; 10: Fiber;
+	# 11: Fault; 12: Link/activity(tx); 13: Link/activity(rx);
+	# 14: Link (master); 15: separate register
+	uci set network.@switch_port[-1].led='6'
+
+	ucidef_add_switch_port "switch0" "2"
+	# Port 2 controls the RED configuration of LEDs for
+	# the switch and the section does not correspond to a real
+	# switch port.
+	#
+	# See the key above for switch port 1 for the meaning of the
+	# 'led' setting below.
+	uci set network.@switch_port[-1].led='9'
+	;;
+
+DGN3500B)
+ 	lan_mac=$(mtd_get_mac_ascii uboot-env ethaddr)
+ 	wan_mac=$(macaddr_add "$lan_mac" 1)
+ 	ucidef_set_interface_lan "eth0.1"
+ 	ucidef_add_switch "switch0" "1" "1"
+ 	ucidef_add_switch_vlan "switch0" "1" "3 2 1 0 5t"
+		# Blinkrate: 0=43ms; 1=84ms; 2=120ms; 3=170ms; 4=340ms; 5=670ms
+	uci set network.@switch[-1].blinkrate='1'
+
+	ucidef_add_switch_port "switch0" "1"
+	# Port 1 controls the GREEN configuration of LEDs for
+	# the switch and the section does not correspond to a real
+	# switch port.
+	#
+	# 0=LED off; 1=Collision/FDX; 2=Link/activity; 3=1000 Mb/s;
+	# 4=100 Mb/s; 5=10 Mb/s; 6=1000 Mb/s+activity; 7=100 Mb/s+activity;
+	# 8=10 Mb/s+activity; 9=10/100 Mb/s+activity; 10: Fiber;
+	# 11: Fault; 12: Link/activity(tx); 13: Link/activity(rx);
+	# 14: Link (master); 15: separate register
+	uci set network.@switch_port[-1].led='6'
+
+	ucidef_add_switch_port "switch0" "2"
+	# Port 2 controls the RED configuration of LEDs for
+	# the switch and the section does not correspond to a real
+	# switch port.
+	#
+	# See the key above for switch port 1 for the meaning of the
+	# 'led' setting below.
+	uci set network.@switch_port[-1].led='9'
+	annex="b"
 	;;
 
 GR7000)
@@ -179,9 +234,9 @@
 [ -z "$(ls /lib/modules/`uname -r`/ltq_atm*)" ] || set_atm_wan "$vpi" "$vci" "$encaps" "$payload"
 
 if [ -n "$(grep "system type.*: VR9" /proc/cpuinfo)" ]; then
-	set_vdsl_modem "$annex" "/lib/firmware/vdsl.bin" "av" "ptm"
+	set_vdsl_modem "$annex" "/lib/firmware/vdsl/vdsl.bin" "av" "ptm"
 else
-	set_adsl_modem "$annex" "/lib/firmware/adsl.bin"
+	set_adsl_modem "$annex" "/lib/firmware/adsl/adsl.bin"
 fi
 
 [ -n "$lan_mac" ] && ucidef_set_interface_macaddr lan $lan_mac
--- trunk/target/linux/lantiq/base-files/etc/uci-defaults/01_leds
+++ trunk/target/linux/lantiq/base-files/etc/uci-defaults/01_leds
@@ -7,17 +7,13 @@
 . /lib/functions/uci-defaults.sh
 . /lib/functions/lantiq.sh
 
-[ -e "/sys/class/leds/wifi" ] && ucidef_set_led_wlan "wifi" "wifi" "wifi" "phy0tpt"
-[ -e "/sys/class/leds/usb" ] && ucidef_set_led_usbdev "usb" "usb" "usb" "1-1"
-[ -e "/sys/class/leds/usb2" ] && ucidef_set_led_usbdev "usb2" "usb2" "usb2" "2-1"
-[ -e "/sys/class/leds/internet" ] && ucidef_set_led_netdev "internet" "internet" "internet" "pppoe-wan"
-
 board=$(lantiq_board_name)
 
 case "$board" in
 VG3503J)
 	ucidef_set_led_netdev "vdsl" "vdsl" "bt:green:dsl" "ptm0"
 	;;
+	
 BTHOMEHUBV2B)
 	ucidef_set_led_default "power" "power" "soc:blue:power" "1"
 #	ucidef_set_led_wlan "wifi" "wifi" "soc:blue:wireless" "phy0radio"
@@ -25,6 +21,7 @@
 	ucidef_set_led_netdev "internet" "internet" "soc:blue:broadband" "nas0"
 	ucidef_set_led_usbdev "usb" "usb" "soc:blue:phone" "1-1"
 	;;
+	
 VGV7519)
 	ucidef_set_led_default "power" "power" "power" "0"
 	ucidef_set_led_default "power2" "power2" "power2" "1"
@@ -32,15 +29,18 @@
 	ucidef_set_led_netdev "internet_yellow" "internet_yellow" "internet_yellow" "nas0"
 	ucidef_set_led_netdev "broadband_yellow" "broadband_yellow" "broadband_yellow" "pppoe-wan"
 	;;
+	
+DGN3500*)
+	ucidef_set_led_default "power_green" "Power Status" "DGN3500:green:power" "1"
+	ucidef_set_led_netdev "wifi_green" "WIFI - TXRX" "DGN3500:green:wifi" "wlan0"
+	ucidef_set_led_netdev "internet_green" "Internet" "DGN3500:green:internet" "nas0"
+	ucidef_set_led_default "dsl" "DSL Status" "DGN3500:dsl" "0"
+	ucidef_set_led_usbdev "usb" "USB Status" "DGN3500:usb" "1-1"
+	;;
 *)
 	;;
 esac
 
-for a in `ls /sys/class/leds/`; do
-	grep -q "\[none\]" /sys/class/leds/$a/trigger
-	[ $? -eq 0 ] && ucidef_set_led_default $a $a $a `cat /sys/class/leds/$a/brightness`
-done
-
 ucidef_commit_leds
 
 exit 0
--- trunk/files/etc/build
+++ trunk/files/etc/build
@@ -0,0 +1,1 @@
+Compiled by hitech95
--- trunk/package/network/services/hostapd/files/wps-hotplug.sh
--- trunk/package/network/services/hostapd/files/wps-hotplug.sh
@@ -1,8 +1,11 @@
 #!/bin/sh
 
 if [ "$ACTION" = "pressed" -a "$BUTTON" = "wps" ]; then
+	echo "255" > /sys/class/leds/DGN3500:wps/brightness
 	for dir in /var/run/hostapd*; do
 		[ -d "$dir" ] || continue
 		hostapd_cli -p "$dir" wps_pbc
 	done
+	sleep 10
+	echo "0" > /sys/class/leds/DGN3500:wps/brightness
 fi
